src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/metrics/QualityScoreDistributionSpark.java

```diff
@@ -0,0 +1,181 @@
+import com.google.common.annotations.VisibleForTesting;
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMReadGroupRecord;
+import htsjdk.samtools.metrics.MetricsFile;
+import htsjdk.samtools.util.Histogram;
+import htsjdk.samtools.util.SequenceUtil;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.api.java.function.Function;
+import org.broadinstitute.hellbender.cmdline.Argument;
+import org.broadinstitute.hellbender.cmdline.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.cmdline.programgroups.SparkProgramGroup;
+import org.broadinstitute.hellbender.engine.spark.GATKSparkTool;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.tools.picard.analysis.QualityScoreDistribution;
+import org.broadinstitute.hellbender.utils.R.RScriptExecutor;
+import org.broadinstitute.hellbender.utils.gcs.BucketUtils;
+import org.broadinstitute.hellbender.utils.io.Resource;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.read.ReadUtils;
+
+import java.io.*;
+import java.util.List;
+
+/**
+ * Charts quality score distribution within a BAM file.
+ *
+ * This is the spark version of the tool.
+ */
+@CommandLineProgramProperties(
+        summary = "Program to chart " +
+                "quality score distributions in a SAM or BAM file.",
+        oneLineSummary = "Charts quality score distributions for a SAM or BAM file",
+        programGroup = SparkProgramGroup.class
+)
+public final class QualityScoreDistributionSpark extends GATKSparkTool {
+    private static final long serialVersionUID = 1l;
+
+    @Argument(doc = "uri for the output file: a local file path",
+            shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME, fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+            optional = true)
+    public String out;
+
+    @Argument(shortName="CHART", doc="A file (with .pdf extension) to write the chart to.", optional=true)
+    public File CHART_OUTPUT;
+
+    @Argument(doc="If set to true calculate mean quality over aligned reads only.")
+    public boolean ALIGNED_READS_ONLY = false;
+
+    @Argument(shortName="PF", doc="If set to true calculate mean quality over PF reads only.")
+    public boolean PF_READS_ONLY = false;
+
+    @Argument(doc="If set to true, include quality for no-call bases in the distribution.")
+    public boolean INCLUDE_NO_CALLS = false;
+
+    @VisibleForTesting
+    static final class Counts implements Serializable{
+        private static final long serialVersionUID = 1L;
+        private final long[] qCounts;
+        private final long[] oqCounts;
+        private final boolean includeNoCalls;
+
+        Counts(final boolean includeNoCalls){
+           qCounts = new long[128];
```

Comment:
1) What's the significance of 128? Put the answer in a comment.
2) Consider moving 128 into a named constant.